File: Diary-AI-FE/frontend-react/src/services/insights.js
import api from './api';

export const insightsAPI = {
  getPersonalized: (days = 90) => api.get(`/api/insights/personalized?days=${days}`),
  getOptimization: (days = 60, metric = 'energy_level') =>
    api.get(`/api/insights/optimization?days=${days}&metric=${metric}`),
};

export default insightsAPI;

File: Diary-AI-BE/app/presentation/routers/insights.py
from __future__ import annotations
from fastapi import APIRouter, Query, HTTPException
from presentation.controllers import insights_controller as ctl

router = APIRouter(tags=["insights"], prefix="/insights")

@router.get("/personalized")
async def personalized_insights(days: int = Query(60, ge=7, le=365)):
    try:
        return await ctl.personalized(days)
    except Exception as e:  # pragma: no cover
        raise HTTPException(status_code=500, detail=str(e))

@router.get("/optimization")
async def optimization_insights(metric: str = Query("sleep_quality"), days: int = Query(60, ge=7, le=365)):
    try:
        return await ctl.optimization(metric, days)
    except Exception as e:  # pragma: no cover
        raise HTTPException(status_code=500, detail=str(e))

__all__ = ["router"]
